js/views/userPage/UserPageFollow.js

```diff
@@ -1,23 +1,104 @@
 
-export default class extends View {
+export default class extends BaseVw {
   constructor(options = {}) {
-    super({
-      className: 'userPageFollow',
-      events: {
-      },
-      ...options,
-    });
+    super(options);
+    this.options = options;
+    this.followType = options.followType;
+
+    if (!options.ownPage) {
+      this.followCol = new Follows(null, {
+        type: this.followType.toLowerCase(),
+        guid: this.model.id,
+      });
+      this.followCol.fetch().done(() => {
+        this[`updateViewer${this.followType}`](true);
+        this.render();
+      });
+
+      this.listenTo(app.ownFollowing, 'sync, update', () => {
+        this.updateViewerFollowers();
+        this.render();
+      });
+
+      this.listenTo(app.ownFollowers, 'update', () => {
+        this.updateViewerFollowing();
+        this.render();
+      });
+    } else {
+      this.followCol = app[`own${this.followType}`];
+      if (this.followType === 'Followers') {
+        this.followCol.fetch().done(() => {
+          this.render();
+        }); // TODO: pagination
+      } else {
+        this.render();
+      }
+
+      this.listenTo(app[`own${this.followType}`], 'sync, update', () => {
+        this.render();
+      });
+    }
   }
 
-  render() {
-    loadTemplate('userPage/userPageFollow.html', (t) => {
-      this.$el.html(t({
-        ...this.model.toJSON(),
-      }));
-    });
+  updateViewerFollowers() {
+    /* if the viewer follows/unfollows this user, add them to the followers list without a fetch */
+    if (this.followType === 'Followers' && !this.options.ownPage) {
+      // if the viewer has followed add them
+      if (followUtils.followedByYou(this.model.id)) {
+        this.followCol.add({ guid: app.profile.id });
+        // if the viewer has unfollowed remove them
+      } else if (!followUtils.followedByYou(this.model.id)) {
+        this.followCol.remove(app.profile.id); // remove by id
+      }
+    }
+  }
 
+  updateViewerFollowing(viaAPI = false) {
+    /* if the viewer is unfollowed or followed, update the following list */
+    if (this.followType === 'Following' && !this.options.ownPage) {
+      let followsYou;
+      if (viaAPI) {
+        // get via isFollowingMe API when it is ready
+        console.log('update via API');
+      } else {
+        followsYou = followUtils.followsYou(this.model.id);
+      }
+      // if this page has followed the viewer add them
+      if (followsYou) {
+        this.followCol.add({ guid: app.profile.id });
+        // if this page has unfollowed the viewer remove them
+      } else if (!followsYou) {
+        this.followCol.remove(app.profile.id); // remove by id
+      }
+    }
+  }
+
+  className() {
+    return 'userPageFollow flexRow';
+  }
+
+  render() {
+    this.$el.empty();
+    // TODO: add pagination if the collection is Followers
+    if (this.followCol.length) {
+      this.followCol.forEach((follow) => {
+        const user = this.createChild(userShort, {
```

Comment:
Some cleanup is necessary here. If on every render you are creating whole new user views (and there might be hundreds or thousands), then you need to store the current ones in a data structure... and on a new render, iterate through and call remove() on the old ones.

Yes, by using 'createChild()', when the tabView is removed, the children will be... but if while on this tab, you re-render multiple time and the list is large, you'll have a whole lot of unused views in memory. It's just safer to clean them up when you know they're not needed anymore.